tests/test_derivatives.py

```diff
@@ -416,9 +416,13 @@ def test_subsampled_fd(self):
 
         eqns = [Eq(u.forward, u + 1.), Eq(u2.forward, u2.dx)]
         op = Operator(eqns)
-        op.apply(time_M=nt-2)
-        # Verify that u2[1, x,y]= du2/dx[0, x, y]
 
+        # NOTE: To assert against u.data[-1] we need to provide explicit bounds
```

Comment:
@mloubout note

Reply:
Not sure I follow why, if these are needed now in such case it's a bit of a pain to have the user needing to infer those.

Reply:
no, not necessary in general.

Just in this case because you don't just care about `u2`, but also about the modulo buffered TimeFunction namely `u`

I could have also changed the assert to

```
assert np.allclose(u.data[-2], nt-2)
```

Iterating up to 11 or 10 is legal -- in both cases `u2` will have tracked its last value.
Whether `u.data[-1]` is of interest or not is up to the user